app/src/main/java/com/github/orkest/View/profile/ProfileTopInterface.kt

```diff
@@ -72,19 +77,51 @@ fun ProfileTopInterface(viewModel: ProfileViewModel) {
 
         Spacer(modifier = Modifier.height(separator))
 
-        Row(){
-            EditButton {
-                val intent = Intent(context, EditProfileActivity::class.java)
-                context.startActivity(intent)
+
+        Row{
+            if(viewModel.username.value == Constants.currentLoggedUser) {
+                EditButton {
+                    val intent = Intent(context, EditProfileActivity::class.java)
+                    context.startActivity(intent)
+                }
+            } else {
+                FollowButton(viewModel, viewModel.isUserFollowed.observeAsState().value)
             }
         }
     }
 }
 
+@Composable
+fun FollowButton(viewModel: ProfileViewModel, isUserFollowed: Boolean?){
+    if(isUserFollowed == null){ Text(text="") } //Empty body, here waiting for the future that fetch isUserFollowed to complete

```

Comment:
We talked about this, I look it up, and that seems like the correct work around.  I don't beleive appart from caching the values, that we can a better work around for a better UI experience

Reply:
Yes, you're right. It's the best option for now